MediaContentSynchronization/Model/Synchronize.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentSynchronization\Model;
+
+use Magento\MediaContentSynchronizationApi\Api\SynchronizeInterface;
+use Magento\MediaContentSynchronizationApi\Model\SynchronizerPool;
+use Magento\Framework\Exception\LocalizedException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Synchronize content with assets
+ */
+class Synchronize implements SynchronizeInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $log;
+
+    /**
+     * @var SynchronizerPool
+     */
+    private $synchronizerPool;
+
+    /**
+     * @param LoggerInterface $log
+     * @param SynchronizerPool $synchronizerPool
+     */
+    public function __construct(
+        LoggerInterface $log,
+        SynchronizerPool $synchronizerPool
+    ) {
+        $this->log = $log;
+        $this->synchronizerPool = $synchronizerPool;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(): void
+    {
+        $failedItems = [];
+
+        /* var $synchronizer SynchronizerInterface */
+        foreach ($this->synchronizerPool->get() as $synchronizer) {
+            try {
+                $synchronizer->execute();
+            } catch (\Exception $exception) {
+                $this->log->critical($exception);
+                $failedItems[] = $exception;
+            }
+        }
+
+        if (!empty($failedItems)) {
+            throw new LocalizedException(
+                __('Could not synchronize content with asset')
```

Comment:
I would be good to provide more information on failed operation, at least failed synchronizer name